pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -419,32 +432,56 @@ def _apply_lightning_controlled(self, operation):
                         operation.base.matrix, control_wires, control_values, target_wires, False
                     )
 
-        def apply_lightning(self, operations):
+        def _apply_lightning_midmeasure(self, operation: MidMeasureMP, mid_measurements: dict):
+            """Execute a MidMeasureMP operation and return the sample in mid_measurements.
+
+            Args:
+                operation (~pennylane.operation.Operation): mid-circuit measurement
+
+            Returns:
+                None
+            """
+            wires = self.wires.indices(operation.wires)
+            wire = list(wires)[0]
+            sample = qml.math.reshape(self.generate_samples(shots=1), (-1,))[wire]
+            if operation.postselect is not None and sample != operation.postselect:
+                mid_measurements[operation] = -1
+                return
```

Comment:
Ok just confirming here, an invalid shot is recorded as a `-1` outcome and the statevector is left untouched right?

Reply:
Correct